### PR TITLE
Pyrsia homebrew installer for macOS

### LINKED ISSUE #1167: Create Homebrew Installer using Ruby
### Suggestion details

- Create Homebrew Installer using Ruby and required directory.
- Test installer in local machine.

### Additional details

_No response_

### PR DESCRIPTION
**Pull request by dasmanas**
## Description

Fixes pyrsia/pyrsia#1167 & pyrsia/pyrsia#1170
This PR takes care the installation of Pyrsia on macOS. Please refer the readme.md to install Pyrsia in your local macOS (intel arch x86) machine. Try out different pyrsia operations from your local machine.

## How to test
brew tap expects the latest is in main branch of the repo. Until and unless the PR is merge we **can't install** it using
```
brew tap pyrsia/pyrsia
brew install pyrsia
```
To overcome this block you have to edit the formula file of Pyrsia in your local using `brew create <url>`.
Replace `<url>` from taking the `url` value from `Formula/pyrsia.rb` file from the commit. This will open the pyrsia.rb in your default editor. Replace entire content of the file from `Formula/pyrsia.rb`. Now run `brew install pyrsia`

### Alternate strategy to test
Test from my personal repo.
```
brew tap dasmanas/pyrsiaanother
brew install pyrsia
```

## PR Checklist

- [x] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [x] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/get_involved/good_pr.md)
- [x] I've included a good title and brief description along with how to review them.
- [x] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).

### Code Contributions

- [x] I've run the unit tests `brew test pyrsia` and everything passes.
- [x] I've made sure my brew is current and updated.

### CODE REVIEW

File: Formula/pyrsia.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  def install
+    ENV.deparallelize
+    bin.install "pyrsia"
+    bin.install "pyrsia_node"
+  end
+
+  test do
+    system "true"
+  end
+end
```
Comment (AbhijithGanesh) [inline Formula/pyrsia.rb:18]: This looks good to me barring the ruby syntax (which I am unfamiliar of)

**Review by AbhijithGanesh — COMMENTED** [review]
LGTM

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# IntelliJ
+.idea
+
+# VSCode
+.vscode
```
Comment (fishseabowl) [inline .gitignore]: Add a newline at end of file

**Review by karolh2000 — APPROVED** [review]
The good news is the binaries work on M1, the bad news is that we were not able to confirm if the brew installer works with M1.

File: .gitignore (hunk 2 above)
Comment (dasmanas) [reply .gitignore]: > Add a newline at end of file

done.

**Comment by dasmanas** [comment]
> The good news is the binaries work on M1, the bad news is that we were not able to confirm if the brew installer works with M1.

Here I can see Homebrew is supported for Apple silicon. Refer https://brew.sh/2021/02/05/homebrew-3.0.0/

**Comment by erwin1** [comment]
I tested the brew install on M1. Works fine 👍  (although i do get x86_64 of course, but it runs)

Instruction should be added to https://github.com/pyrsia/pyrsia/blob/main/docs/tutorials/quick-installation.mdx - I'll create an issue to track this

**Pull request merged**